### ISSUE TITLE
Waking up from suspend sometimes does not work on kevin device

### ISSUE THREAD

**Issue by WizzardSK** [Maccraft123/Cadmium#40]
Using Ubuntu Hirsute with KDE, kernel 5.14rc4cadmium, running from SD card.
Sometimes when I suspend my chromebook, I am not able to wake it up, I have to turn it off and on.
[dmesg.txt](https://github.com/Maccraft123/Cadmium/files/6948968/dmesg.txt)

**Comment by drocsid** [comment]
@WizzardSK  can you add test cases here?

**Comment by WizzardSK** [comment]
I am not sure what you mean. I try suspend, mostly it works, sometimes it does not.

**Comment by drocsid** [comment]
The idea is that you describe a way to reproduce your bug.

So you can trigger a suspend like:

sudo systemctl suspend

Then the expectation is that you do something and it resumes. That way the bug fixer has a test case.

**Comment by WizzardSK** [comment]
I run the command: sudo systemctl suspend OR I close the lid.
Mostly I am able to resume the OS, after opening the lid or pressing some key. maybe 5 times it works and OS resumes, but sometimes not.

**Comment by drocsid** [comment]
Fair enough! I'll let the chromium devs take this up, until I get involved

**Comment by Maccraft123** [comment]
> Fair enough! I'll let the cadmium devs take this up, until I get involved

yo i'm on holidays

**Comment by citral23** [comment]
That's not a mature and acceptable answer, please go back to work.

**Comment by drocsid** [comment]
@WizzardSK I think maybe you can capture logs which contain the suspend cycle, for a functioning and non-functioning run. I do not know where to look at the top of my head. Can you look into this?

**Comment by WizzardSK** [comment]
This is how it looks when not working:
Aug 10 22:07:29 gru kernel: [ 2406.317604] PM: suspend entry (deep)
(hard reset by me)
Aug 10 22:08:13 gru kernel: [    0.000000] Booting Linux on physical CPU 0x0000000000 [0x410fd034]

This is when it works:
Aug 10 21:26:48 gru kernel: [  607.170709] PM: suspend entry (deep)
Aug 10 21:26:48 gru kernel: [  607.198499] Filesystems sync: 0.027 seconds
Aug 10 21:37:31 gru kernel: [  607.199729] Freezing user space processes ... (elapsed 0.002 seconds) done.
Aug 10 21:37:31 gru kernel: [  607.202640] OOM killer disabled.
Aug 10 21:37:31 gru kernel: [  607.202644] Freezing remaining freezable tasks ... (elapsed 0.144 seconds) done.
Aug 10 21:37:31 gru kernel: [  607.347276] printk: Suspending console(s) (use no_console_suspend to debug)
Aug 10 21:37:31 gru kernel: [  607.349144] mwifiex_pcie 0000:01:00.0: None of the WOWLAN triggers enabled
Aug 10 21:37:31 gru kernel: [  607.697347] Disabling non-boot CPUs ...
Aug 10 21:37:31 gru kernel: [  607.699171] psci: CPU1 killed (polled 1 ms)
Aug 10 21:37:31 gru kernel: [  607.701601] psci: CPU2 killed (polled 1 ms)
Aug 10 21:37:31 gru kernel: [  607.703579] psci: CPU3 killed (polled 1 ms)
Aug 10 21:37:31 gru kernel: [  607.705031] IRQ 25: no longer affine to CPU4
Aug 10 21:37:31 gru kernel: [  607.705596] psci: CPU4 killed (polled 1 ms)
Aug 10 21:37:31 gru kernel: [  607.707653] psci: CPU5 killed (polled 1 ms)
Aug 10 21:37:31 gru kernel: [  607.710341] Enabling non-boot CPUs ...
Aug 10 21:37:31 gru kernel: [  607.713242] Detected VIPT I-cache on CPU1
Aug 10 21:37:31 gru kernel: [  607.713581] GICv3: CPU1: found redistributor 1 region 0:0x00000000fef20000
Aug 10 21:37:31 gru kernel: [  607.714078] CPU1: Booted secondary processor 0x0000000001 [0x410fd034]
Aug 10 21:37:31 gru kernel: [  607.718724] CPU1 is up
Aug 10 21:37:31 gru kernel: [  607.719927] Detected VIPT I-cache on CPU2
Aug 10 21:37:31 gru kernel: [  607.719975] GICv3: CPU2: found redistributor 2 region 0:0x00000000fef40000
Aug 10 21:37:31 gru kernel: [  607.720179] CPU2: Booted secondary processor 0x0000000002 [0x410fd034]
Aug 10 21:37:31 gru kernel: [  607.722454] CPU2 is up
Aug 10 21:37:31 gru kernel: [  607.723585] Detected VIPT I-cache on CPU3
Aug 10 21:37:31 gru kernel: [  607.723642] GICv3: CPU3: found redistributor 3 region 0:0x00000000fef60000
Aug 10 21:37:31 gru kernel: [  607.723800] CPU3: Booted secondary processor 0x0000000003 [0x410fd034]
Aug 10 21:37:31 gru kernel: [  607.726252] CPU3 is up
Aug 10 21:37:31 gru kernel: [  607.727714] Detected PIPT I-cache on CPU4
Aug 10 21:37:31 gru kernel: [  607.727840] GICv3: CPU4: found redistributor 100 region 0:0x00000000fef80000
Aug 10 21:37:31 gru kernel: [  607.728189] CPU4: Booted secondary processor 0x0000000100 [0x410fd082]
Aug 10 21:37:31 gru kernel: [  607.733497] CPU4 is up
Aug 10 21:37:31 gru kernel: [  607.734402] Detected PIPT I-cache on CPU5
Aug 10 21:37:31 gru kernel: [  607.734437] GICv3: CPU5: found redistributor 101 region 0:0x00000000fefa0000
Aug 10 21:37:31 gru kernel: [  607.734541] CPU5: Booted secondary processor 0x0000000101 [0x410fd082]
Aug 10 21:37:31 gru kernel: [  607.736490] CPU5 is up
Aug 10 21:37:31 gru kernel: [  607.813180] [drm:analogix_dp_resume] *ERROR* failed to setup the panel
Aug 10 21:37:31 gru kernel: [  607.814600] PM: dpm_run_callback(): rockchip_dp_resume+0x0/0x2c returns -16
Aug 10 21:37:31 gru kernel: [  607.814889] rockchip-dp ff970000.edp: PM: failed to resume early: error -16
Aug 10 21:37:31 gru kernel: [  608.068926] usb 1-1: reset high-speed USB device number 2 using ehci-platform
Aug 10 21:37:31 gru kernel: [  608.476012] usb usb5: root hub lost power or was reset
Aug 10 21:37:31 gru kernel: [  608.476421] usb usb6: root hub lost power or was reset
Aug 10 21:37:31 gru kernel: [  608.482068] usb usb7: root hub lost power or was reset
Aug 10 21:37:31 gru kernel: [  608.482188] usb usb8: root hub lost power or was reset
Aug 10 21:37:31 gru kernel: [  608.818565] OOM killer enabled.
Aug 10 21:37:31 gru kernel: [  608.818646] Restarting tasks ... done.
Aug 10 21:37:31 gru kernel: [  608.927880] PM: suspend exit
